### PR TITLE
Second review round on #388: guard direction, lint gates, MCP gate logging

### PR DESCRIPTION
**Pull request by DonaldMurillo**
Follow-up to #388. The review bot posted a second review on that PR a few minutes after its last push, ten of the eleven findings outside the diff, and the PR merged before they were triaged. All eleven hold on merit and are fixed here, each pinned by a fixture that fails without the change:

- **GOFASTR1006**: constants that conflict within one directory (build-tag variants) become unresolved rather than one overwriting the other.
- **divlimit and intwrap**: selector divisors match by binding identity, not formatted text; a guard counts only when the reaching path is proven safe (an unsafe-range comparison must diverge, a safe-range comparison must enclose the use), so `if limit == 0 { return total / limit }` fires again.
- **Runtime lints**: a truthy `.status` is not a response gate; the attribute gate must dominate the URL construction; the escape call must be the whole selector operand; an identifier plus a number is numeric only when the identifier is proven numeric; regex escapes that can match a separator are rejected; member assignments are not identifier assignments and parameters shadow file-scope safe names; function scopes match on the blanked view so a `}` inside a regex literal cannot close one.
- **MCP gates**: a recovered panic is logged with the operation and tool name before the internal error is returned.

Verification: `go test` on the analyzers, contracts, lint, and MCP packages, the runtime security probes, `make analyze`, and `gofastr verify --no-vet --strict` clean on the tree.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_014BLjmqxTmrgQgZg2S2wefZ

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved static-analysis accuracy for runtime shapes, URL construction, numeric conversions, division checks, and prefix boundaries.
  - Reduced false positives and false negatives by accounting for control-flow branches, variable shadowing, build variants, escaping, and response-status validation.
  - Improved recognition of safe guards and rejecting branches in security-related checks.

- **Security**
  - Recovered gate failures are now logged with relevant operation context while continuing to return internal errors.

- **Tests**
  - Expanded coverage for analyzer edge cases, security-gate failures, build variants, escaping, status checks, and control-flow behavior.

### CODE REVIEW

File: core-ui/check/runtimeshapes_test.go [hunk 1]
```diff
@@ -1061,3 +1061,275 @@ func TestLintAttributePathSegments_Review5GatesAndSanitizers(t *testing.T) {
...
+// function.
+const holder = {};
+holder.id = 'fixed';
+function usePlain(id) {
+  return document.querySelector('[data-plain="' + id + '"]');
+}
+const id = 'fixed';
+function useParam(id) {
+  return document.querySelector('[data-param="' + id + '"]');
+}
```
[the remark this answers is not in the document]
Comment (DonaldMurillo) [reply core-ui/check/runtimeshapes_test.go:1309]: Accepted, fixed in 36b3f70d with a fixture that fails without the change.

File: core-ui/check/runtimeshapes.go [hunk 2]
```diff
@@ -1633,6 +1835,105 @@ func responseGateIn(span string) bool {
...
+				lit, _ := strconv.Atoi(num[1])
+				switch {
+				case op[1] == "===" || op[1] == "==" || op[1] == "!==" || op[1] == "!=":
+					if lit >= 200 && lit <= 299 {
+						return true
+					}
+				case op[1] == "<" || op[1] == "<=":
+					if lit <= 399 {
+						return true
+					}
```
[the remark this answers is not in the document]
Comment (DonaldMurillo) [reply core-ui/check/runtimeshapes.go:1873]: Accepted, fixed in 36b3f70d with a fixture that fails without the change.

File: core-ui/check/runtimeshapes.go [hunk 3]
```diff
@@ -2144,8 +2571,11 @@ func regexGateAnchored(lit string) bool {
 	if !strings.HasPrefix(body, "^") || !strings.HasSuffix(body, "$") {
 		return false
 	}
-	for _, forbidden := range []string{".", `\s`, `\S`, `\W`, "[^"} {
-		if strings.Contains(body, forbidden) {
+	for _, r := range body[1 : len(body)-1] {
+		if r < 0x20 || r == 0x7f {
+			return false // control characters
+		}
+		if !strings.ContainsRune(regexSafeClassChars, r) {
```
[the remark this answers is not in the document]
Comment (DonaldMurillo) [reply core-ui/check/runtimeshapes.go:2597]: Accepted, fixed in 36b3f70d with a fixture that fails without the change.

File: core/mcp/listgate_security_test.go [hunk 4]
```diff
@@ -656,3 +658,40 @@ func TestServerGateClosesResourcesSurface(t *testing.T) {
...
+
+	if err := runCallGate(func(string) error { panic("call gate boom") }, "kiln_probe"); err == nil || err.Error() != "internal tool error" {
+		t.Fatalf("runCallGate: panicking gate must still answer the internal tool error, got %v", err)
+	}
+	logged := buf.String()
+	if !strings.Contains(logged, "call gate boom") {
+		t.Errorf("runCallGate: recovered panic not logged: %q", logged)
+	}
+	if !strings.Contains(logged, "kiln_probe") {
+		t.Errorf("runCallGate: log must carry the tool name: %q", logged)
```
[the remark this answers is not in the document]
Comment (DonaldMurillo) [reply core/mcp/listgate_security_test.go:681]: Accepted, fixed in 36b3f70d with a fixture that fails without the change.

File: internal/analyzers/divlimit/divlimit.go [hunk 5]
```diff
@@ -118,73 +126,243 @@ func checkFunc(pass *analysis.Pass, fnType *ast.FuncType, body *ast.BlockStmt) {
...
+	case *ast.UnaryExpr:
+		if x.Op == token.NOT {
+			return condProvesNonzero(pass, x.X, divisor, !holds)
 		}
-		if isZeroOrOne(pass, other) {
-			return true
+	case *ast.BinaryExpr:
+		switch x.Op {
+		case token.LAND:
+			return condProvesNonzero(pass, x.X, divisor, holds) || condProvesNonzero(pass, x.Y, divisor, holds)
```
[the remark this answers is not in the document]
Comment (DonaldMurillo) [reply internal/analyzers/divlimit/divlimit.go]: Accepted, fixed in 36b3f70d with a fixture that fails without the change.
